barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.cpp

```diff
@@ -221,55 +294,775 @@ GoblinTranslatorCircuitBuilder::AccumulationInput generate_witness_values(
+    // Split z scalars into microlimbs for range constraining
+    for (size_t i = 0; i < GoblinTranslatorCircuitBuilder::NUM_Z_LIMBS - 1; i++) {
         input.z_1_microlimbs[i] = split_standard_limb_into_micro_limbs(input.z_1_limbs[i]);
         input.z_2_microlimbs[i] = split_standard_limb_into_micro_limbs(input.z_2_limbs[i]);
     }
-    for (size_t i = 0; i < GoblinTranslatorCircuitBuilder::NUM_BINARY_LIMBS; i++) {
+    input.z_1_microlimbs[GoblinTranslatorCircuitBuilder::NUM_Z_LIMBS - 1] = split_top_z_limb_into_micro_limbs(
+        input.z_1_limbs[GoblinTranslatorCircuitBuilder::NUM_Z_LIMBS - 1], TOP_Z_MICROLIMB_BITS);
+    input.z_2_microlimbs[GoblinTranslatorCircuitBuilder::NUM_Z_LIMBS - 1] = split_top_z_limb_into_micro_limbs(
+        input.z_2_limbs[GoblinTranslatorCircuitBuilder::NUM_Z_LIMBS - 1], TOP_Z_MICROLIMB_BITS);
+
+    // Split current accumulator into microlimbs for range constraining
+    for (size_t i = 0; i < last_limb_index; i++) {
         input.current_accumulator_microlimbs[i] = split_standard_limb_into_micro_limbs(input.current_accumulator[i]);
-        // info("Stored: ", single_accumulation_step.current_accumulator_microlimbs[i][5], " at ", i);
     }
-    for (size_t i = 0; i < GoblinTranslatorCircuitBuilder::NUM_BINARY_LIMBS; i++) {
+    input.current_accumulator_microlimbs[last_limb_index] =
+        split_top_limb_into_micro_limbs(input.current_accumulator[last_limb_index], TOP_STANDARD_MICROLIMB_BITS);
+
+    // Split quotient into microlimbs for range constraining
+    for (size_t i = 0; i < last_limb_index; i++) {
         input.quotient_microlimbs[i] = split_standard_limb_into_micro_limbs(input.quotient_binary_limbs[i]);
-        // info("Stored: ", single_accumulation_step.current_accumulator_microlimbs[i][5], " at ", i);
     }
+    input.quotient_microlimbs[last_limb_index] =
+        split_top_limb_into_micro_limbs(input.quotient_binary_limbs[last_limb_index], TOP_QUOTIENT_MICROLIMB_BITS);
     return input;
 }
-template GoblinTranslatorCircuitBuilder::AccumulationInput generate_witness_values(
-    barretenberg::fr op_code,
-    barretenberg::fr p_x_lo,
-    barretenberg::fr p_x_hi,
-    barretenberg::fr p_y_lo,
-    barretenberg::fr p_y_hi,
-    barretenberg::fr z_1,
-    barretenberg::fr z_2,
-    barretenberg::fq previous_accumulator,
-    barretenberg::fq v,
-    barretenberg::fq x);
+/**
+ * @brief Create a single accumulation gate
+ *
+ * @param acc_step
+ */
+void GoblinTranslatorCircuitBuilder::create_accumulation_gate(const AccumulationInput acc_step)
+{
+    // The first wires OpQueue/Transcript wires
+    // Opcode should be {0,1,2,3,4,8}
+    ASSERT(acc_step.op_code == 0 || acc_step.op_code == 1 || acc_step.op_code == 2 || acc_step.op_code == 3 ||
+           acc_step.op_code == 4 || acc_step.op_code == 8);
+
+    auto& op_wire = std::get<WireIds::OP>(wires);
+    op_wire.push_back(add_variable(acc_step.op_code));
+    // Odd op values are not defined so let's just put zero there
```

Comment:
The word odd might be misleading here since I think what you're really referring to is the second row in a pair corresponding to a single operation

Reply:
Explained in code